File: app/src/main/java/com/pluscubed/velociraptor/api/osm/OsmApiEndpoint.java
package com.pluscubed.velociraptor.api.osm;

public class OsmApiEndpoint {
    public final String baseUrl;
    //Wait 1 minute before trying a slow/errored endpoint again.
    public long timeTakenTimestamp;
    public int timeTaken;
    //null if public
    public String name;

    public OsmApiEndpoint(String baseUrl, String name) {
        this.baseUrl = baseUrl;
        this.name = name;
    }

    @Override
    public String toString() {
        String time;
        if (timeTaken == Integer.MAX_VALUE) {
            time = "error";
        } else if (timeTaken == 0) {
            time = "pending";
        } else {
            time = timeTaken + "ms";
        }

        String title;
        if (name != null) {
            title = name;
        } else {
            title = this.baseUrl;
        }

        return title + " - " + time + ", " + timeTakenTimestamp;
    }
}

File: app/src/main/java/com/pluscubed/velociraptor/api/osm/OsmApiEndpoint.kt
package com.pluscubed.velociraptor.api.osm

class OsmApiEndpoint(val baseUrl: String) : Comparable<OsmApiEndpoint> {

    var service: OsmService? = null
        get() {
            if (field == null) {
                throw Exception("OSM Service not set")
            }
            return field
        }
    var timeTaken: Int = 0

    override fun toString(): String {
        val time: String
        if (timeTaken == Integer.MAX_VALUE) {
            time = "error"
        } else if (timeTaken == 0) {
            time = "pending"
        } else {
            time = timeTaken.toString() + "ms"
        }

        return this.baseUrl + " - " + time
    }

    override fun compareTo(other: OsmApiEndpoint): Int {
        return timeTaken.compareTo(other.timeTaken);
    }
}
